tests/specs/mcd/verification.k

```diff
@@ -130,6 +125,33 @@ module LEMMAS-MCD-HASKELL [symbolic, kore]
        andBool Y <=Int 0
        andBool notBool #rangeUInt(256, X -Int Y) [simplification]
 
+    rule { chop(X) #Equals chop(Y) } => { X #Equals Y }
+         requires #rangeSInt(256, X) andBool #rangeSInt(256, Y)
+         [simplification]
+
+    rule { chop ( X:Int ) #Equals chop ( Y:Int ) } =>
+         { true #Equals X ==Int Y andBool Y <=Int 0 }
+         requires (minSInt256 <=Int X) andBool (X <=Int 0) andBool #rangeSInt(256, Y) [simplification]
```

Comment:
I believe that if the second lemma can apply, the first one can too? Why do we need both? If we have both, we'll be relying on non-deterministic application order of the backend, which is probably not good.

Could we instead have this lemma talk about just `{ X #Equals Y }`, so we let the first lemma do it's work then handle what it leaves behind?

Reply:
I will remove the second one and see what happens with the relevant tests.

Reply:
They pass, the second one is not needed.

Reply:
Unresolving until the code shows up here.

Reply:
Should be there now.